test/tools/TestExe/TestExe.cs

```diff
@@ -40,6 +47,48 @@ static void EchoArgs(string[] args)
             }
         }
 
+        static void WriteStandardOutput()
+        {
+            // write a string of characters (bytes), this closest resembles a binary
+            // write the string "test" with an accent over the e
+            byte[] testbytes = new byte[] { 116, 233, 115, 116 };
+            foreach(byte b in testbytes)
+            {
+                Console.Write((char)b);
+            }
+        }
+
+        static void ReadStandardInput(string[] args)
+        {
+            Encoding encodingToUse = null;
+            if (args.Length > 1)
+            {
+                // the encoding page to get could be provided as an argument;
+                int result;
+                if ( int.TryParse(args[1], out result))
+                {
+                    try
+                    {
+                        encodingToUse = Encoding.GetEncoding(result);
+                    }
+                    catch
+                    {
+                        ;
+                    }
+                }
+            }
+            if ( encodingToUse == null )
+            {
+                encodingToUse = Encoding.GetEncoding(28591);
+            }
+            Console.InputEncoding = encodingToUse;
+            string pipedText = Console.In.ReadToEnd();
```

Comment:
Maybe we should just spit out the characters read in without having the encoding involved? For example, like this:
```
int ch;
while ((ch = Console.In.Read()) != -1)
{
    Console.WriteLine(ch);
}
```

Reply:
I like that much better, thanks
updated